Marlin/Configuration.h

```diff
@@ -921,6 +921,10 @@
 //#define MANUAL_Y_HOME_POS 0
 //#define MANUAL_Z_HOME_POS 0
 
+// If enabled, the global workspace offset will not include the Z home offset.
+// This is useful if you want to set a home position using M206 without adjusting the soft limits and the probing system
```

Comment:
Since `M206` shifts the current position, the endstops need to be adjusted to keep them in the same relative position, so the physical movement constraints stay the same. If `home_offset` isn't included in the software endstop calculation, and the home offset is set to a perceptible value (>5mm), then movement will be unusually constrained by the software endstops, and the nozzle will not be allowed to move in its previous physical range.

The `workspace_offset` is also used when figuring the leveling compensation. Again, if `home_offset` is not included in `workspace_offset` then the leveling mesh will not be aligned to the current position. It will be offset by the `home_offset`.

Was your aim to only affect the Z home offset (as implied above)? If so, this PR has too many other side-effects and in fact does break some things. The `workspace_offset` is used throughout the code, so the XY home offset has to be considered. We can't adopt a feature that breaks the workspace behavior.

It sounds like you just want to set your soft endstops out a little farther, so they don't get in the way when you make small adjustments to your `home_offset`. That is perfectly allowable already, and doesn't come with added side-effects.